Modules/socketmodule.c

```diff
@@ -2684,6 +2684,7 @@ sock_setblocking(PySocketSockObject *s, PyObject *arg)
     if (block == -1 && PyErr_Occurred())
         return NULL;
 
+    block = !!block;
```

Comment:
I suggest to add instead a new int variable. For example, rename "long block" to "long value", and add a new "int block = (value != 0);"?

Reply:
sock_settimeout() already uses "int block = timeout < 0;"

By the way, this code in internal_setblocking() is ugly :-(

```
    block = !block;
    if (ioctl(s->sock_fd, FIONBIO, (unsigned int *)&block) == -1)
```

I'm not sure that it's a good idea to cast int argument to (unsigned int*). Maybe adding a new "unsigned int block_ui = block;" would be safer?

Reply:
Regarding `internal_setblocking`, its Linux part duplicates `_Py_set_blocking` in `Python/fileutils.c`, which doesn't have the same cast. So I'd just call `_Py_set_blocking` directly instead, but not in this PR if this is to be backported to 2.7.

`ioctl` is a var-arg function per [POSIX](http://pubs.opengroup.org/onlinepubs/9699919799/functions/ioctl.html). The cast was introduced in 0e3f591aeeef9ed715f8770320f4c4c7332a8794, but I didn't dig further on. Since there were no complains about `_Py_set_blocking`, I suspect it's obsolete.